parse/parser.go

```diff
@@ -219,12 +249,11 @@ func parseIdentifier(buf TokenBuffer) (ast.Expression, []error) {
 		errs = append(errs, errors.New("parseIdentifier() - "+token.Val+" is not a identifier"))
 		return nil, errs
 	}
-	buf.Read()
 	return &ast.Identifier{Value: token.Val}, nil
 }
 
 func parseIntegerLiteral(buf TokenBuffer) (ast.Expression, []error) {
-	token := buf.Read()
+	token := buf.Peek(CURRENT)
```

Comment:
I think `Read()` is write. because when we execute this parse function we assume that current token is `IntegerLiteral`

Reply:
Ok, Instead of, I should remove buf.Read() in makeInfixExpression()